flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamGraphGeneratorTest.java

```diff
@@ -806,6 +809,54 @@ public void testConflictSlotSharingGroup() {
         env.getStreamGraph();
     }
 
+    @Test
+    public void testTrackTransformationsByIdentity() {
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final Transformation<?> noopTransformation = env.fromSequence(1, 2).getTransformation();
+
+        final StreamGraphGenerator generator =
+                new StreamGraphGenerator(
+                        Lists.newArrayList(
+                                noopTransformation,
+                                new FailingTransformation(noopTransformation.hashCode())),
```

Comment:
Is there a benefit here using Guava instead of `Arrays.asList`?

Reply:
I do not think but having one dependency less is always favorable. I used `Arrays.asList`.